File: src/main/java/za/ac/cput/service/CustomerService.java
package za.ac.cput.service;

import java.util.List;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import za.ac.cput.domain.UType.Customer;
import za.ac.cput.repository.CustomerRepository;

@Service
public class CustomerService implements ICustomerService {

    private final CustomerRepository repository;

    @Autowired
    public CustomerService(CustomerRepository repository) {
        this.repository = repository;
    }

    @Override
    public Customer create(Customer customer) {
        return repository.save(customer);
    }

    @Override
    public Customer read(UUID id) {
        return repository.findById(id).orElse(null);
    }

    @Override
    public Customer update(Customer customer) {
        if (repository.existsById(customer.getUserId())) {
            return repository.save(customer);
        }
        return null;
    }

    @Override
    public boolean delete(UUID id) {
        repository.deleteById(id);
        return !repository.existsById(id);
    }

    @Override
    public List<Customer> getAll() {
        return repository.findAll();
    }

    @Override
    public List<Customer> findByPaymentMethod(String paymentMethod){
        return repository.findByPaymentMethod(paymentMethod);
        }

    @Override
    public Customer login(String email, String password) {
        return repository.findByEmailAndPassword(email, password);
    }

    @Override
    public Customer findByEmail(String email) {
        return repository.findByEmail(email);
    }
}

File: Frontend/src/api/customerService.ts
// src/api/customerService.ts
import axiosInstance from './axiosInstance';

export interface Customer {
    id?: string;
    firstName: string;
    lastName: string;
    email: string;
    password?: string;
    paymentMethod?: string;
}

export const registerCustomer = async (customerData: Customer) => {
    const response = await axiosInstance.post('/customer/create', customerData);
    return response.data;
};

export const loginCustomer = async (credentials: { email: string; password: string }) => {
    const response = await axiosInstance.post('/customer/login', credentials);
    return response.data;
};

export const getCustomer = async (id: string) => {
    const response = await axiosInstance.get(`/customer/read/${id}`);
    return response.data;
};

export const updateCustomer = async (customerData: Customer) => {
    const response = await axiosInstance.put('/customer/update', customerData);
    return response.data;
};

export const deleteCustomer = async (id: string) => {
    const response = await axiosInstance.delete(`/customer/delete/${id}`);
    return response.data;
};

export const getAllCustomers = async () => {
    const response = await axiosInstance.get('/customer/getAll');
    return response.data;
};

export const findCustomersByPaymentMethod = async (paymentMethod: string) => {
    const response = await axiosInstance.get(`/customer/findByPaymentMethod?paymentMethod=${encodeURIComponent(paymentMethod)}`);
    return response.data;
};